### PR TITLE
Add UNETR++

### PR DESCRIPTION
**Pull request by innat**
fix https://github.com/innat/medic-ai/issues/7

## Summary by CodeRabbit

* **New Features**
  * Added UNETR++ model, new UNet/UNETR layer blocks, Efficient Paired Attention, and a uniform initializer.

* **API Changes**
  * Standardized public block names and constructor params (e.g., filters), switched skip wiring to list-based inputs, exported new model symbols, and added a dropout-layer helper.

* **Documentation**
  * Added UNETR++ docs and updated README.

* **Tests**
  * Added UNETR++ tests and updated dice metric tests (param renamed to target_class_ids).

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: medicai/layers/attention.py [hunk 1]
```diff
@@ -161,3 +163,213 @@ def get_config(self):
...
+
+        # Transpose for consistency with Channel Attention operations
+        q_shared_t = ops.transpose(q_shared, (0, 1, 3, 2))
+        k_shared_t = ops.transpose(k_shared, (0, 1, 3, 2))
+        v_CA_t = ops.transpose(v_CA, (0, 1, 3, 2))
+        v_SA_t = ops.transpose(v_SA, (0, 1, 3, 2))
+
+        # Normalize q and k - l2 norm
+        q_norm = self.safe_normalize(q_shared_t, axis=-1, epsilon=1e-6)
+        k_norm = self.safe_normalize(k_shared_t, axis=-1, epsilon=1e-6)
```
[the remark this answers is not in the document]
Comment (innat) [reply medicai/layers/attention.py]: /gemini

The EPA block performs Channel Attention to calculate the similarity between different channels (heads). If we normalized across axis 2, we would be normalizing the pixel feature vector, which is useful for Spatial attention but would make the Channel attention dot product mathematically meaningless.

If we look at the official UNETR++ PyTorch implementation, the tensor manipulation order confirms axis=-1

```python
# From official UNETR++ (simplified)
# 1. Start with (B, H, N, D)
q_shared = q_shared.transpose(-2, -1) 
# 2. Now shape is (B, H, D, N)

# 3. Normalize dim=-1
q_shared = torch.nn.functional.normalize(q_shared, dim=-1)
# Since the last dimension is N, they are normalizing the sequence length.
```

While standard Self-Attention normalizes the feature dimension to compute token-to-token similarity, the EPA block in UNETR++ computes Channel-to-Channel similarity. To treat the channels as vectors for Cosine Similarity, we must normalize across the spatial dimension. This matches the official UNETR++ implementation where normalize(dim=-1) is applied to the (B, H, D, N) tensor.
